game/dota_addons/dota_imba/scripts/vscripts/hero/hero_shadow_shaman.lua

```diff
@@ -0,0 +1,203 @@
+
+LinkLuaModifier("modifier_imba_mass_serpent_ward",  "hero/hero_shadow_shaman", LUA_MODIFIER_MOTION_NONE)
+
+-- TODO: make count as hero damage
+
+function imba_shadow_shaman_mass_serpent_ward:OnSpellStart()
+    -- Ability properties
+    local caster            =   self:GetCaster()
+    local spawn_point       =   self:GetCursorPosition()
+    local ward_level        =   self:GetLevel()
+    local ward_name         =   "npc_imba_dota_shadow_shaman_ward_"
+    local response          =   "shadowshaman_shad_ability_ward_"
+    local spawn_particle    =   "particles/units/heroes/hero_shadowshaman/shadowshaman_ward_spawn.vpcf"
+    -- Ability parameters
+    local ward_count    =   self:GetSpecialValueFor("ward_count")
+    local ward_duration =   self:GetSpecialValueFor("duration")
+
+    -- Emit cast sound
+    caster:EmitSound("Hero_ShadowShaman.SerpentWard")
+    -- Emit cast response
+    caster:EmitSound(response..RandomInt(4,7))
+
+    -- Emit spawn particle
+    local spawn_particle_fx = ParticleManager:CreateParticle(spawn_particle, PATTACH_ABSORIGIN, caster)
+    ParticleManager:SetParticleControl( spawn_particle_fx, 0, spawn_point )
+
+    -- Create 8 wards in a box formation (note the vectors)
+    ward_table  =   {}
+    ward_table[1] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(-64, -64 ,0), true, caster, caster, caster:GetTeamNumber())
+    ward_table[2] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(-64, 0 ,0), true, caster, caster, caster:GetTeamNumber())
+    ward_table[3] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(-64, 64 ,0), true, caster, caster, caster:GetTeamNumber())
+    ward_table[4] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(0, -64 ,0), true, caster, caster, caster:GetTeamNumber())
+    ward_table[5] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(0, 64 ,0), true, caster, caster, caster:GetTeamNumber())
+    ward_table[6] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(64, -64 ,0), true, caster, caster, caster:GetTeamNumber())
+    ward_table[7] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(64, 0 ,0), true, caster, caster, caster:GetTeamNumber())
+    ward_table[8] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(64, 64 ,0), true, caster, caster, caster:GetTeamNumber())
+    -- Create 2 additional wards next to the other ones
+    ward_table[9] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(-128 , 64 ,0), true, caster, caster, caster:GetTeamNumber())
+    ward_table[10] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(128 , 64 ,0), true, caster, caster, caster:GetTeamNumber())
+
+    -- #1 TALENT: More Serpent Wards
+    if caster:HasTalent("special_bonus_imba_shadow_shaman_1") then
+        ward_table[11] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(-128, -64 ,0), true, caster, caster, caster:GetTeamNumber())
```

Comment:
Can include this in another `vecs` array and treat the same way as my previous example.